src/data_provider/src/osinfo/sysOsParsers.cpp

```diff
@@ -137,15 +137,28 @@ static bool findVersionInStream(std::istream& in,
 bool UnixOsParser::parseFile(std::istream& in, nlohmann::json& info)
 {
     constexpr auto SEPARATOR{'='};
-    static const std::map<std::string, std::string> KEY_MAP
+    bool ret;
+
+    static const std::map<std::string, std::string> KEY_MAP1
     {
         {"NAME",             "os_name"},
         {"VERSION",          "os_version"},
         {"ID",               "os_platform"},
         {"BUILD_ID",         "os_build"},
         {"VERSION_CODENAME", "os_codename"}
     };
-    const auto ret {parseUnixFile(KEY_MAP, SEPARATOR, in, info)};
+    ret = parseUnixFile(KEY_MAP1, SEPARATOR, in, info);
+
+    if (info.find("os_version") == info.end())
+    {
+        static const std::map<std::string, std::string> KEY_MAP2
+        {
+            {"VERSION_ID",       "os_version"},
+        };
+        in.clear();
+        in.seekg(0);
+        ret |= parseUnixFile(KEY_MAP2, SEPARATOR, in, info);
```

Comment:
What is the purpose of bitwise operations in boolean? couldn't this lead to unexpected behavior?

Reply:
Just to avoid a conditional statement to set variable ret=true in case function parseUnixFile returns true.
In C++ boolean variables are guaranteed to be false=0 and true=1 in case of integer promotion.
![image](https://user-images.githubusercontent.com/101227434/216365076-66277094-b67d-46d8-99e3-28f0e3cd067e.png)